### PR TITLE
Remove NuGet package path support in CBT.NuGet and CBT.NuGet.GlobalRe…

### PR DESCRIPTION
**Pull request by jeffkl**
…store

Users will be required to specify a path in their NuGet.config for packages.config.

Fix package property generation in Visual Studio with an InitialTargets target

Update AggregatePackages module as well

### CODE REVIEW

File: src/CBT.NuGet/Internal/NuGetProjectJsonParser.cs [hunk 1]
```diff
@@ -1,53 +1,90 @@
...
             LockFile lockFile = LockFileUtilities.GetLockFile(lockFilePath, NullLogger.Instance);
 
-            VersionFolderPathResolver versionFolderPathResolver = new VersionFolderPathResolver(packagesPath);
+            string globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(NuGetSettings);
 
-            foreach (LockFileLibrary library in lockFile.Libraries)
+            if (String.IsNullOrWhiteSpace(globalPackagesFolder))
             {
-                yield return new PackageIdentityWithPath(library.Name, library.Version, versionFolderPathResolver.GetPackageDirectory(library.Name, library.Version), versionFolderPathResolver.GetInstallPath(library.Name, library.Version));
+                throw new NuGetConfigurationException(@"Unable to determine the NuGet repository path.  This usually defaults to ""%UserProfile%\.nuget\packages"", ""%NUGET_PACKAGES%"", or the ""globalPackagesFolder"" in your NuGet.config.");
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetProjectJsonParser.cs]: > throw  [](start = 15, length = 7)

Be consistent warning is probably sufficient.

File: src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs [hunk 2]
```diff
@@ -1,85 +1,125 @@
...
-                yield break;
+                return false;
             }
-            // This assumes that if it is a non packages.config or project.json being restored that it is a MSBuild project using the new PackageReference.  
-            if (ProjectJsonPathUtilities.IsProjectConfig(packageConfigPath) || packageConfigPath.EndsWith(Constants.PackageReferenceFile, StringComparison.OrdinalIgnoreCase))
+
+            if (packageRestoreData == null)
             {
-                yield break;
+                Log.LogWarning("Missing expected assets file directory.  This is typically because the flag generated at $(CBTModuleNuGetAssetsFlagFile) does not exist or is empty.  Ensure the GenerateModuleAssetFlagFile target is running. It may also be because the CBTModules.proj does not import CBT build.props in some fashion.");
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: >LogWarning [](start = 20, length = 10)

log a message?

File: src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs [hunk 3]
```diff
@@ -1,85 +1,125 @@
...
             }
-            if (string.IsNullOrWhiteSpace(packageRestoreData?.RestoreOutputAbsolutePath))
+
+            if (String.Equals("Unknown", packageRestoreData?.RestoreProjectStyle, StringComparison.OrdinalIgnoreCase))
             {
-                _log.LogMessage(MessageImportance.Low, "Missing assets file directory.  If you are running NuGet prior to 4.x this is expected.  If you are running NuGet 4.x or higher then this is typically because the flag generated at $(CBTNuGetAssetsFlagFile) does not exist or is empty.  Ensure the GenerateNuGetAssetFlagFile target is running. It may also be because the project does not import cbt build.props in some fashion.  It may also be because NuGet failed to parse the project you may set the env NUGET_RESTORE_MSBUILD_ARGS to the value '/flp:v=diag;logfile=restore.log;append' and check the restore.log for more details. ");
-                yield break;
+                packages = Enumerable.Empty<PackageIdentityWithPath>();
+
+                return true;
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: >true [](start = 23, length = 4)

false?

Comment (kingerja) [reply src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: and comment that this is ordering specific

---
In reply to: [144648337](https://github.com/CommonBuildToolset/CBT.Modules/pull/182#discussion_r144648337) [](ancestors = 144648337)

File: src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs [hunk 4]
```diff
@@ -1,68 +1,80 @@
...
+
+            _configParsersLazy = new Lazy<List<INuGetPackageConfigParser>>(() =>
+            {
+                settings = settings ?? Settings.LoadDefaultSettings(Path.GetDirectoryName(_packageConfigPaths[0]), configFileName: null, machineWideSettings: new XPlatMachineWideSetting());
+
+                return new List<INuGetPackageConfigParser>
+                {
+                    new NuGetPackagesConfigParser(settings, _logger),
+                    new NuGetProjectJsonParser(settings, _logger),
+                    new NuGetPackageReferenceProjectParser(settings, _logger)
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs]: >new  [](start = 20, length = 4)

add comment about ordering

File: src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs [hunk 5]
```diff
@@ -1,68 +1,80 @@
...
+                        // Consider adding item metadata of packageid and version for ease of consumption of this property.
+                        itemGroup.AddItem("CBTNuGetPackageDir", packageInfo.FullPath);
+                    }
+                }
             }
-            project.Save(outputPath);
 
-            return true;
-        }
+            if (propertyGroup.Properties.Count > 1)
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs]: >1 [](start = 49, length = 1)

add comment on why 1

File: src/CBT.NuGet/Tasks/GenerateNuGetProperties.cs [hunk 6]
```diff
@@ -157,12 +145,21 @@ public bool Execute(string packageRestoreFile, string[] inputs, string propsFile
         /// <returns> a PackageRestoreData object based off the contents of the flag file.</returns>
         internal PackageRestoreData GetPackageRestoreData()
         {
-            if (string.IsNullOrWhiteSpace(AssetsFile) || !File.Exists(AssetsFile))
+            if (NuGetPackagesConfigParser.IsPackagesConfigFile(PackageRestoreFile))
             {
-                Log.LogMessage(MessageImportance.Low, $"Package reference {AssetsFile} not found.  Either you are not using PackageReference elements for your packages or you are using a version of nuget.exe prior to 4.x, or the project does not import CBT in some way.");
+                return new PackageRestoreData
+                {
+                    RestoreProjectStyle = "PackagesConfig"
```
Comment (kingerja) [inline src/CBT.NuGet/Tasks/GenerateNuGetProperties.cs]: >RestoreProjectStyle = "PackagesConfig" [](start = 20, length = 38)

Investigate restore style

**Comment by kingerja** [comment]
> <?xml version="1.0" encoding="utf-8"?>

rev to 2.1

---
Refers to: src/CBT.NuGet.AggregatePackage/build/After.CBT.NuGet.props:1 in 9155f37. [](commit_id = 9155f373d6485cffb8c97097b730b1fd46d7794b, deletion_comment = False)

File: src/CBT.NuGet.GlobalRestore/build/After.Microsoft.Common.targets [hunk 7]
```diff
@@ -46,13 +44,12 @@
     <CallTarget Targets="_CheckForCBTNuGetGlobalPackagesRestoredMarker" />
 
     <GenerateNuGetProperties
-      Condition=" '$(CBTNuGetGeneratePackageProperties)' == 'true' And '$(IsRestoreOnly)' == 'true' "
```
Comment (kingerja) [inline src/CBT.NuGet.GlobalRestore/build/After.Microsoft.Common.targets]: >d '$(IsRestoreOnly)' == 'tr [](start = 69, length = 27)

double check this is needed

File: src/CBT.NuGet.Package/version.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://raw.githubusercontent.com/AArnott/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
-  "version": "2.0-dev",
+  "version": "2.1-dev",
```
Comment (kingerja) [inline src/CBT.NuGet.Package/version.json]: >version [](start = 3, length = 7)

rev assembly versions

File: src/CBT.NuGet.Package/build/After.Traversal.targets [hunk 9]
```diff
@@ -7,13 +7,14 @@
     <CBTNuGetTraversalPackagePropertyFile Condition=" '$(CBTNuGetTraversalPackagePropertyFile)' == '' ">$(CBTNuGetTraversalRestoreFile).props</CBTNuGetTraversalPackagePropertyFile>
     <CBTNuGetAllProjects>$(MSBuildProjectFullPath);$(MSBuildThisFileFullPath);$(CBTNuGetAllProjects)</CBTNuGetAllProjects>
     <CBTNuGetTraversalRestoreGlobalProperties Condition=" '$(CBTNuGetTraversalRestoreGlobalProperties)' == '' ">BuildingInsideVisualStudio=true;DesignTimeBuild=true;CBTModulesRestored=true;CBTNuGetGeneratePackageProperties=false;CBTNuGetTraversalPackagePropertiesCreated=true;CBTNuGetTraversalPackagesRestored=true</CBTNuGetTraversalRestoreGlobalProperties>
+    <CBTNuGetTraversalPackagesRestoredMarker Condition=" '$(CBTNuGetTraversalPackagesRestoredMarker)' == '' ">$(CBTNuGetIntermediateOutputPath)\$(MSBuildProjectFile).CBTNuGetTraversalPackagesRestored</CBTNuGetTraversalPackagesRestoredMarker>
```
Comment (kingerja) [inline src/CBT.NuGet.Package/build/After.Traversal.targets]: >CBTNuGetTraversalPackagesRestoredMarker [](start = 5, length = 39)

wtf

File: src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs [hunk 10]
```diff
@@ -1,85 +1,125 @@
...
-                    .Where(lib => lib.Name.Equals(pkg.Id, StringComparison.OrdinalIgnoreCase) &&
-                                  lib.Version.ToString().Equals(pkg.Version, StringComparison.OrdinalIgnoreCase)).Select(lib => lib.Dependencies).SelectMany(p => p.Select(i => i));
+                throw new NuGetConfigurationException(@"Unable to determine the NuGet repository path.  This usually defaults to ""%UserProfile%\.nuget\packages"", ""%NUGET_PACKAGES%"", or the ""globalPackagesFolder"" in your NuGet.config.");
+            }
+
+            globalPackagesFolder = Path.GetFullPath(globalPackagesFolder);
+
+            if (!Directory.Exists(globalPackagesFolder))
+            {
+                throw new DirectoryNotFoundException($"The NuGet repository '{globalPackagesFolder}' does not exist.  Ensure that NuGet is restore packages to the location specified in your NuGet.config.");
```
Comment (kingerja) [inline src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: >is restore [](start = 136, length = 10)

restored

File: src/CBT.NuGet/Internal/NuGetProjectJsonParser.cs (hunk 1 above)
Comment (jeffkl) [reply src/CBT.NuGet/Internal/NuGetProjectJsonParser.cs]: I'm throwing in all tree now

File: src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs (hunk 2 above)
Comment (jeffkl) [reply src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: Done

File: src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs (hunk 3 above)
Comment (jeffkl) [reply src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs]: Done

File: src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs (hunk 4 above)
Comment (jeffkl) [reply src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs]: Done

File: src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs (hunk 5 above)
Comment (jeffkl) [reply src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs]: I introduced a variable to make it more apparent

File: src/CBT.NuGet.Package/version.json (hunk 8 above)
Comment (jeffkl) [reply src/CBT.NuGet.Package/version.json]: Done

File: src/CBT.NuGet/Tasks/GenerateNuGetProperties.cs (hunk 6 above)
Comment (jeffkl) [reply src/CBT.NuGet/Tasks/GenerateNuGetProperties.cs]: Its `PackageReference`, `ProjectJson`, or `Unknown`

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L387-L392

**Pull request merged**